File: java/Seventh/src/arraylist/Student.java
package arraylist;

import java.util.ArrayList;

public class Student {

    int studentID;
    String studentName;
    ArrayList<Subject> subjectList;                 // ArrayList 선언
    int index;

    public Student(int studentID, String studentName) {

        this.studentID = studentID;
        this.studentName = studentName;

        subjectList = new ArrayList<Subject>();     // ArrayList 생성
    }

    public void addSubject(String name, int score) {

        Subject subject = new Subject();        // Subject 생성

        subject.setName(name);                  // 과목 이름 추가
        subject.setScorePoint(score);           // 점수 추가
        subjectList.add(subject);               // 배열에 저장
    }

    public void showStudentInfo() {

        int total = 0;

        for (Subject s : subjectList) {

            total += s.getScorePoint();         // 총점 더하기

            System.out.println("학생 " + studentName + "의 " + s.getName() + " 과목 성적은 " + s.getScorePoint() + " 입니다.");
        }

        System.out.println("학생 " + studentName + "의 총점은 " + total + " 입니다.");
    }
}

File: python/p_oop2/chap06/array/student.py
from subject import Subject

class Student:
    def __init__(self, student_id, student_name):
        self.student_id = student_id
        self.student_name = student_name
        self.subject_list = []  # 과목에 대한 객체 정보를 저장하기 위한 리스트

    def add_subject(self, name, score):
        subject = Subject(name, score)
        self.subject_list.append(subject)  # 과목 리스트에 과목의 이름과 점수를 객체로 저장

    def show_student_info(self):
        total = 0
        for subject in self.subject_list:
            total += subject.get_score()
            print(self.student_name + " 학생의 " + subject.get_name() + " 과목의 성적은 " \
            + str(subject.get_score()) + "점 입니다.");
        print(self.student_name + " 학생의 총점은 " + str(total) + "점 입니다.")
